### PR TITLE
migrate `examples` to `openApi` part 3; affects [conan cookbook coverity cpan debian docker docsrs dub eclipse]

### PR DESCRIPTION
**Pull request by chris48s**
Refs #9285

..and so begins the long slow process of working through these. The first batch I am working on is classes with:

- currently passing service tests
- only path params
- either one example, or two examples due to optional path params (e.g: `branch*`)
- no documentation string

These are the most trivial services to convert, requiring the least manual work.
I am going to submit PRs touching batches of about 15-20 files at a time as I work through these.
Then I'll start moving onto slightly more complicated cases.

### CODE REVIEW

File: services/depfu/depfu.service.js [hunk 1]
```diff
@@ -13,16 +18,23 @@ class Depfu extends BaseJsonService {
     pattern: ':vcsType(github|gitlab)/:project+',
```
Comment (chris48s) [inline services/depfu/depfu.service.js:18]: TODO: enum

Comment (chris48s) [reply services/depfu/depfu.service.js]: I've just reverted this for now

**Comment by calebcartwright** [comment]
I just had a thought that I wish I'd had sooner. I know we've had some discussions about potential options to create helper functions that can return the objects needed for the open api spec, as well as some of the tradeoffs around helper functions/abstractions vs. explicit spec definitions.

However, I feel like we've been discussing that option mostly in the context of routes, and I cant' remember whether we've discussed something similar for the route parameters. Specifically, should we consider adjusting the path/route param helper functions to accept the same form of key/value pairs we've used historically as opposed to each class having to explicitly designate `name` and `example` fields with the respective values?

I don't feel terribly strongly, but it seems like the type of thing we should decide for at least the near to mid future before we get through too many of these conversions

**Comment by calebcartwright** [comment]
E.g. instead of:

```js
pathParams({
  name: 'packageName',
  example: 'boost',
})
```

perhaps:
```js
pathParams({ packageName: 'boost' })
```

**Comment by chris48s** [comment]
So that works in the case where we only care about `name` and `example`, but there are other keys in this object: `schema` and `description`.

In https://github.com/badges/shields/pull/9437 (probably the next one to review tbh) we begin using `schema`.

Currently we don't use `description` because documentation for individual params was not a feature we previously had so all of this is just jammed in the top-level `documentation`/`description`. I'd like to start using it to move documentation about individual params out of the top-level `documentation`/`description` key and onto the relevant params as we go though.

I did use this feature when I set up the new docs for the dynamic badges though. For example https://shields.io/badges/dynamic-json-badge

Screenshot at 2023-08-16 21-22-35

**Comment by calebcartwright** [comment]
Fair enough, it feels like it's more verbose than it could be, but I can also live with it given the schema being broader than what I'd seen/anticipated

**Review by calebcartwright — APPROVED** [review]

**Pull request merged**